python/meterpreter/meterpreter.py

```diff
@@ -507,23 +519,40 @@ def deactivate(self):
 		self.communication_last = 0
 		return True
 
+	def decrypt_packet(self, pkt):
+		if pkt and len(pkt) > PACKET_HEADER_SIZE:
+			# We don't support AES encryption yet, so just do the normal
+			# XOR thing and move on
+			xor_key = struct.unpack('BBBB', pkt[:PACKET_XOR_KEY_SIZE])
+			raw = xor_bytes(xor_key, pkt)
+			return raw[PACKET_HEADER_SIZE:]
+		return None
+
 	def get_packet(self):
 		self.request_retire = False
 		try:
-			pkt = self._get_packet()
+			pkt = self.decrypt_packet(self._get_packet())
 		except:
 			return None
 		if pkt is None:
 			return None
 		self.communication_last = time.time()
 		return pkt
 
+	def encrypt_packet(self, pkt):
+		# The packet now has to contain session GUID and encryption flag info
+		# And given that we're not yet supporting AES, we're going to just
+		# always return the session guid and the encryption flag set to 0
+		# TODO: we'll add encryption soon!
+		xor_key = rand_xor_key()
+		raw = SESSION_GUID + '\x00' + pkt
```

Comment:
This needs to be `raw = bytes(SESSION_GUID, 'UTF-8') + NULL_BYTE + pkt` for Python3 compatibility. There's an additional error that's causing packet IO to fail with Python3 as well that I'm still tracking down.